### PR TITLE
Add goimports CI check and format exiting imports

### LINKED ISSUE #2628: Add go-imports format CI check and format all the existing go-imports
- We need a precheck in GitHub workflows to validate the go-imports format. 
- Format the existing go-imports

### PR DESCRIPTION
**Pull request by specter25**
- Add Check in github actions workflows to check for goimports format
- Format existing go imports
- This fixes #2628 

Signed-off-by: specter25 <ujjwalcoding012@gmail.com>

## Proposed changes

We need a precheck in GitHub workflows to validate the go-imports format.
Format the existing go-imports

## Types of changes

What types of changes does your code introduce to Litmus? Put an `x` in the boxes that apply
- [x] New feature (non-breaking change which adds functionality)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices applies)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
- [x] I have read the [CONTRIBUTING](https://github.com/litmuschaos/litmus/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the commit for DCO to be passed.
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added necessary documentation (if appropriate)

## Dependency
- Please add the links to the dependent PR need to be merged before this (if any).

### CODE REVIEW

**Comment by specter25** [comment]
@rajdas98 sir i have tried to fix the issue #2628 . can you give it a look

**Comment by rajdas98** [comment]
Thanks, @specter25 . Can you comment on that issue?

Regarding the PR: Is this checking all go-related files?

**Comment by specter25** [comment]
@rajdas98 yes sir , it's checking all the go related files .

**Review by rajdas98 — APPROVED** [review]

**Comment by rajdas98** [comment]
LGTM, Let's wait for the other reviewers.

**Comment by specter25** [comment]
sure sir , Thanks :)

**Review by Jasstkn — APPROVED** [review]
LGTM

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -35,6 +35,17 @@ jobs:
...
+  goimports-checks:
+    runs-on: ubuntu-latest
+    needs: changes
+    if: needs.changes.outputs.graphql-server == 'true' || needs.changes.outputs.authentication == 'true' || needs.changes.outputs.authentication == 'true' || needs.changes.outputs.event-tracker == 'true' || needs.changes.outputs.subscriber == 'true'
+    steps:
+    - uses: actions/checkout@v2
+    - name: Check Golang imports order
+      uses: Jerome1337/goimports-action@v1.0.3
+      with:
+        goimports-path: ./
```
Comment (gdsoumya) [inline .github/workflows/build.yml:47]: should be `./litmus-portal` ?

**Review by gdsoumya — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -35,6 +35,17 @@ jobs:
...
+
   backend-checks:
```
Comment (gdsoumya) [inline .github/workflows/build.yml:49]: Maybe we can add the go imports step to the backend checks, I think that would make more sense

**Review by gdsoumya — CHANGES_REQUESTED** [review]

**Comment by gdsoumya** [comment]
Also need to add this step to `push.yaml`

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -35,6 +35,17 @@ jobs:
           subscriber:
             - 'litmus-portal/cluster-agents/subscriber/**'
 
+  goimports-checks:
+    runs-on: ubuntu-latest
+    needs: changes
+    if: needs.changes.outputs.graphql-server == 'true' || needs.changes.outputs.authentication == 'true' || needs.changes.outputs.authentication == 'true' || needs.changes.outputs.event-tracker == 'true' || needs.changes.outputs.subscriber == 'true'
```
Comment (Jonsy13) [inline .github/workflows/build.yml]: Here, the check for output of the authentication filter is repeated. PTAL

File: .github/workflows/build.yml (hunk 2 above)
Comment (Jonsy13) [reply .github/workflows/build.yml]: Yes, it can be added as a step in the backend-checks job. A new job won't be required.

**Review by Jonsy13 — CHANGES_REQUESTED** [review]

**Comment by specter25** [comment]
@gdsoumya sir i have done the changes can you review them once

**Review by Jonsy13 — APPROVED** [review]

**Review by gdsoumya — APPROVED** [review]
LGTM

**Comment by specter25** [comment]
@rajdas98 sir , can the pr be merged now , do are there any more changes ?

**Comment by specter25** [comment]
Thanks sir ;) was nice to contribute .

**Comment by rajdas98** [comment]
@specter25 github workflow failed for push-pipeline. Can you take a look.

**Comment by rajdas98** [comment]
Also, remove the go-import checks from push-pipeline, as we are already checking in build-pipeline

**Comment by specter25** [comment]
Yes @rajdas98  sir , think after i created the pr , some commits were merge before the pr got merged , which were not formatted properly . I will fix it . SO should i create a new pr , or do it in the same branch ?

**Pull request merged**